### ISSUE TITLE
Error in the loss function's implementation

### ISSUE THREAD

**Issue by abdelnour13** [ksgr5566/UnSeGArmaNet#3]
I guess there's an error in the implementation of the [DMON loss](https://github.com/ksgr5566/UnSeGArmaNet/blob/main/gnn_pool.py) the line : `collapse_reg_term = (torch.sqrt(k)/n) * (torch.norm(torch.sum(C.t(), dim=0), p='fro')) - 1` shoudl be  : `collapse_reg_term = (torch.sqrt(k)/n) * (torch.norm(torch.sum(C, dim=0), p='fro')) - 1` without transposing the matrix C.

**Comment by hourlass** [comment]
After reviewing the code, I found that it lacks the most crucial part, the training code. Could you please share how you used it for work on other datasets?

**Comment by ksgr5566** [comment]
@abdelnour13, you can refer to the formula of DMON loss present [here](https://arxiv.org/pdf/2006.16904). We implemented the same for our model.

**Comment by ksgr5566** [comment]
@hourlass, we don't train the model on any particular dataset as this is an unsupervised approach.
